### DISCUSSION TITLE
Pretix-passbooks colors

### DISCUSSION THREAD

**Discussion by fabm3n** [pretix/pretix#3138] [Q&A] [answered]
Hi @wiffbi 

you committed the color stuff for pretix-passbooks. Is there a reason why you are using HEX and not RGB in the pass.json?
In the Apple Documentation I found the following (for example): "A foreground color for the pass, specified as a CSS-style RGB triple, such as rgb(100, 10, 110)." https://developer.apple.com/documentation/walletpasses/pass

I'm asking because I have a issue with white text on a white background image:
![image](https://user-images.githubusercontent.com/11352551/221689105-4f436917-338e-441e-b79d-88893f2238d3.png)

The Label color is working, but the foreground color isn't.
When I remove the background image, it works without any issue:
![image](https://user-images.githubusercontent.com/11352551/221690621-a4aa5157-65ae-4a1e-89af-9e9331f07948.png)

My Colors from the pass.json:
```
    "backgroundColor": "#00ff81",
    "foregroundColor": "#ff0000",
    "labelColor": "#000cff",
```

Could the HEX Codes produce the issue, or is this an Apple Bug?

Best regards
Fabian

**Comment by wiffbi** [comment, marked as answer]
Hi Fabian,

HEX-colors is just what our internal colorpicker-component returns when used in pretix-control. As it works with no background, I do not think it is an RGB/HEX issue. Maybe Apple wallet does some magic on trying to make sure the foregroundColor/background-image contrast is good (according to docs it ignores backgroundColor when an image is used) and decides to overwrite foregroundColor. Could you provide an example passbook-file with background-image so I can have a look and play with it? Thanks!

**Reply by fabm3n** [reply]
Here is an example: https://1drv.ms/u/s!AkO1OuGSOKbMioNCYBhnKmUtAneJVg?e=DkrHae

**Reply by wiffbi** [reply]
Thank you very much for the example. I did some tests myself and changed the background-image to images with solid colors of different lightness As soon as a background-image is set, Wallet on iOS ignores foregroundColor and sets it to white, no matter how light the background-color is (I even added a pure white background and it changed foregroundColor to white as well). Interestingly though I could not find any info on this behaviour online. I do not see any way to change this behaviour through a setting. So basically you either need to use a dark background-image or no background-image at all.
